File: 14. Longest Common Prefix/Solution.py
class Solution:
    def longestCommonPrefix(self, strs):
        """
        :type strs: List[str]
        :rtype: str
        """
        if not strs:
            return ""
        minlen = len(strs[0])
        for i in range(1, len(strs)):
            minlen = len(strs[i]) if len(strs[i]) < minlen else minlen
        if minlen == 0:
            return ""
        ans = ""
        for i in range(minlen):
            for j in range(len(strs) - 1):
                if strs[j][i] != strs[j + 1][i]:
                    return ans
            ans += strs[0][i]
        return ans

File: 14. Longest Common Prefix/Solution.cpp
#include <string>
#include <vector>
#include <iostream>
#include <unordered_map>
using namespace std;
class Solution
{
  public:
    string longestCommonPrefix(vector<string> &strs)
    {
        if (!strs.size())
            return "";
        int minlen = strs[0].size();
        for (int i = 1; i < strs.size(); i++)
            minlen = strs[i].size() < minlen ? strs[i].size() : minlen;
        if (!minlen)
            return "";
        string ans("");
        for (int i = 0; i < minlen; i++)
        {
            for (int j = 0; j < strs.size() - 1; j++)
            {
                if (strs[j][i] != strs[j + 1][i])
                    return ans;
            }
            ans += strs[0][i];
        }
        return ans;
    }
};